provider/rancher.go

```diff
@@ -320,6 +323,27 @@ func listRancherServices(client *rancher.RancherClient) []*rancher.Service {
 	return servicesList
 }
 
+func listRancherPublicEndponts(client *rancher.RancherClient) map[string]string {
+
+	var publicEndpointMap = make(map[string]string)
+
+	hosts, err := client.Host.List(nil)
+
+	if err != nil {
+		log.Errorf("Cannot get Rancher Endpoints %+v", err)
+	}
+
+	for k := range hosts.Data {
+		for e := range hosts.Data[k].PublicEndpoints {
+			endpointData := hosts.Data[k].PublicEndpoints[e]
+			endpoint := endpointData.(map[string]interface{})
+			publicEndpointMap[endpoint["instanceId"].(string)] = endpoint["ipAddress"].(string)
```

Comment:
Is there a way to avoid the runtime type casting somehow?

If not, we should consider doing proper error handling here by checking whether the assertion succeeded.